### PR TITLE
Rework cloning of assets for simulation

### LINKED ISSUE #371: Simulator doesn't honor "ignore keywords in this flow"
Checking the box on the flow doesn't seem to affect the simulator, keywords will still work.

### PR DESCRIPTION
**Pull request by rowanseymour**
Fixes https://github.com/nyaruka/mailroom/issues/371

### CODE REVIEW

File: core/handlers/session_triggered_test.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func TestSessionTriggered(t *testing.T) {
 					assert.Equal(t, []models.ContactID{models.GeorgeID}, start.ContactIDs())
 					assert.Equal(t, []models.GroupID{models.TestersGroupID}, start.GroupIDs())
 					assert.Equal(t, simpleFlow.ID(), start.FlowID())
-					assert.JSONEq(t, `{"parent_uuid":"36284611-ea19-4f1f-8611-9bc48e206654", "ancestors":1, "ancestors_since_input":1}`, string(start.SessionHistory()))
+					assert.JSONEq(t, `{"parent_uuid":"39a9f95e-3641-4d19-95e0-ed866f27c829", "ancestors":1, "ancestors_since_input":1}`, string(start.SessionHistory()))
```
Comment (rowanseymour) [inline core/handlers/session_triggered_test.go:86]: only cuz we're doing less UUID generation

File: core/models/assets.go [hunk 2]
```diff
@@ -413,31 +410,46 @@ func (a *OrgAssets) FieldByKey(key string) *Field {
 	return a.fieldsByKey[key]
 }
 
-// Clone clones our org assets, returning a copy that can be modified without affecting the main
-func (a *OrgAssets) Clone(ctx context.Context, db *sqlx.DB) (*OrgAssets, error) {
+// CloneForSimulation clones our org assets for simulation
+func (a *OrgAssets) CloneForSimulation(ctx context.Context, db *sqlx.DB, newDefs map[assets.FlowUUID]json.RawMessage, testChannels []assets.Channel) (*OrgAssets, error) {
```
Comment (rowanseymour) [inline core/models/assets.go:414]: passing the extra assets in here feels simpler and also means not rebuilding session assets multiple times

**Review by nicpottier — APPROVED** [review]

File: core/models/assets.go [hunk 3]
```diff
@@ -413,31 +410,46 @@ func (a *OrgAssets) FieldByKey(key string) *Field {
...
+		// make a clone of the flow with the provided definition
+		cf := &Flow{}
+		cf.f.UUID = flowUUID
+		cf.f.ID = f.f.ID
+		cf.f.Name = f.f.Name
+		cf.f.Config = f.f.Config
+		cf.f.FlowType = f.f.FlowType
+		cf.f.Version = f.f.Version
+		cf.f.IgnoreTriggers = f.f.IgnoreTriggers
+		cf.f.Definition = newDef
```
Comment (nicpottier) [inline core/models/assets.go]: Now seeing this maybe feels like this should be a private method on Flow but not a big deal.

Comment (rowanseymour) [reply core/models/assets.go]: updated

**Pull request merged**